src/components/Secrets/SecretForm.tsx

```diff
@@ -112,19 +120,26 @@ const SecretForm: React.FC<React.PropsWithChildren<SecretFormProps>> = ({ existi
         onSelect={(_, value: string) => {
           if (isPartnerTask(value, optionsValues)) {
             void setFieldValue('keyValues', [
-              ...values.keyValues.filter((kv) => !kv.readOnlyKey && (!!kv.key || !!kv.value)),
+              ...values.opaque.keyValues.filter(
+                (kv) => !kv.readOnlyKey && (!!kv.key || !!kv.value),
+              ),
               ...getSupportedPartnerTaskKeyValuePairs(value, optionsValues),
             ]);
           }
           void setFieldValue('secretName', value);
         }}
       />
-      <KeyValueFileInputField
-        name="keyValues"
-        data-test="secret-key-value-pair"
-        entries={defaultKeyValues}
-        disableRemoveAction={values.keyValues.length === 1}
-      />
+      {currentType === SecretTypeDropdownLabel.source && <SourceSecretForm />}
+      {currentType !== SecretTypeDropdownLabel.source && (
+        <KeyValueFileInputField
+          required
+          name={
+            currentType === SecretTypeDropdownLabel.opaque ? 'opaque.keyValues' : 'image.keyValues'
+          }
+          entries={defaultKeyValues}
+          disableRemoveAction={values.opaque.keyValues.length === 1}
+        />
+      )}
```

Comment:
you need to test this conditional rendering to make the codecov happy.

Reply:
added by mocking the component